addons/purchase/models/purchase.py

```diff
@@ -156,7 +156,8 @@ def name_get(self):
     @api.model
     def create(self, vals):
         if vals.get('name', 'New') == 'New':
-            vals['name'] = self.env['ir.sequence'].next_by_code('purchase.order') or '/'
+            company_id = vals.get('company_id', self.default_get(['company_id'])['company_id'])
+            vals['name'] = self.env['ir.sequence'].with_context(force_company=company_id).next_by_code('purchase.order') or '/'
```

Comment:
Why not putting the company in context here to have the two fixes in only one create() ?

Reply:
Well, this fix is nearly one year old, but I suppose I did so to avoid having misleading and/or useless content in the context (when purchase_stock is not installed).

It reduces the potential side-effects of the bugfix by specifying the context key only in the module where it's used.

Reply:
I see it's an old one but imo we should send the company even if `purchase_stock` is not installed. You can check `currency_id` has the same problematic default function than the picking_type one.